server/data/nomisClientBuilder.js

```diff
@@ -80,12 +91,17 @@ module.exports = token => {
       return prisoners.map(addReleaseDatesToPrisoner)
     },
 
-    async getOffenderSentencesByNomisId(nomisIds) {
+    async getOffenderSentencesByNomisId(nomisIds, batchSize = 50) {
       const path = `${apiUrl}/offender-sentences`
-      const query = { offenderNo: nomisIds }
-      const headers = { 'Page-Limit': 10000 }
+      if (isEmpty(nomisIds)) {
+        return []
+      }
+
+      const prisoners = await batchRequests(nomisIds, batchSize, batch => {
+        const query = { offenderNo: batch }
+        return nomisGet({ path, query, headers: { 'Page-Limit': batchSize } })
+      })
```

Comment:
I just did something similar for pathfinder, and for the same endpoint! It seemed pretty happy with much larger batch sizes than the default 50 here. In PF it was more than happy at 500 - but just informational - no problem with batch sizes lower than this.